internal/fleet/package_resource.go

```diff
@@ -0,0 +1,120 @@
+package fleet
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients/fleet"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+)
+
+func getPackageID(name, version string) string {
+	hash, _ := utils.StringToHash(name + version)
+
+	return *hash
+}
+
+func ResourcePackage() *schema.Resource {
+	packageSchema := map[string]*schema.Schema{
+		"name": {
+			Description: "The package name.",
+			Type:        schema.TypeString,
+			Required:    true,
```

Comment:
Should the name/version have `ForceNew: true`?

Reply:
Yeah that makes sense, I'll add it.